docs/src/userguide/extension_types.rst

```diff
@@ -553,72 +618,24 @@ usually set on an extension type using a decorator:
 
             @cython.final
             cdef class Parrot:
-               def done(self): pass
-
-Trying to create a Python subclass from this type will raise a
-:class:`TypeError` at runtime.  Cython will also prevent subtyping a
-final type inside of the same module, i.e. creating an extension type
-that uses a final type as its base type will fail at compile time.
-Note, however, that this restriction does not currently propagate to
-other extension modules, so even final extension types can still be
-subtyped at the C level by foreign code.
-
-
-C methods
-=========
-
-Extension types can have C methods as well as Python methods. Like C
-functions, C methods are declared using
-
-* :keyword:`cdef` instead of :keyword:`def` or ``@cfunc`` decorator for *C methods*, or
-* :keyword:`cpdef` instead of :keyword:`def` or ``@ccall`` decorator for *hybrid methods*.
-
-C methods are "virtual", and may be overridden in derived
-extension types. In addition, :keyword:`cpdef`/``@ccall`` methods can even be overridden by Python
-methods when called as C method. This adds a little to their calling overhead
-compared to a :keyword:`cdef`/``@cfunc`` method:
-
-.. tabs::
-
-    .. group-tab:: Pure Python
-
-        .. literalinclude:: ../../examples/userguide/extension_types/pets.py
+               def describe(self): pass
 
-    .. group-tab:: Cython
 
-        .. literalinclude:: ../../examples/userguide/extension_types/pets.pyx
-
-.. code-block:: text
-
-    # Output
-    p1:
-    This parrot is resting.
-    p2:
-    This parrot is resting.
-    Lovely plumage!
-
-The above example also illustrates that a C method can call an inherited C
-method using the usual Python technique, i.e.::
-
-    Parrot.describe(self)
-
-:keyword:`cdef`/``@ccall`` methods can be declared static by using the ``@staticmethod`` decorator.
-This can be especially useful for constructing classes that take non-Python
-compatible types:
-
-.. tabs::
-
-    .. group-tab:: Pure Python
+            @cython.cclass
+            cdef class Lizard:
 
-        .. literalinclude:: ../../examples/userguide/extension_types/owned_pointer.py
 
-    .. group-tab:: Cython
+               @cython.final
+               cdef done(self): pass
 
-        .. literalinclude:: ../../examples/userguide/extension_types/owned_pointer.pyx
-
-.. note::
+Trying to create a Python subclass from a final type or overriding a final method will raise
+a :class:`TypeError` at runtime.  Cython will also prevent subtyping a
+final type or overriding a final method inside of the same module, i.e. creating
+an extension type that uses a final type as its base type will fail at compile time.
+Note, however, that this restriction does not currently propagate to
+other extension modules, so even final extension types can still be
+subtyped at the C level by foreign code.
```

Comment:
```suggestion
other extension modules, so Cython is unable to prevent final extension types
from being subtyped at the C level by foreign code.
```

The existing wording implied that "inheriting from final types from foreign code is a supported feature". I've suggested a wording that's more like "we can't stop you, unfortunately"